R/default_aes.R

```diff
@@ -0,0 +1,174 @@
+#' Loaded in to `cmapplot_globals` in `.onLoad`.
+#'
+#' This list's names MUST equal `cmapplot_globals$geoms_that_change`
+#'
+#' @noRd
+init_cmap_default_aes <- function () {
+  list(
+    Line = list(
+      size = gg_lwd_convert(cmapplot_globals$consts$lwd_plotline)
+    ),
+    Text = list(
+      family = cmapplot_globals$font$strong$family,
+      fontface = ifelse(cmapplot_globals$font$strong$face == "bold", 2, 1),
+      size = cmapplot_globals$fsize$M/ggplot2::.pt, # pt-to-mm conversion
+      colour = cmapplot_globals$colors$blackish
+    ),
+    TextLast =list(
+      family = cmapplot_globals$font$strong$family,
+      fontface = ifelse(cmapplot_globals$font$strong$face == "bold", 2, 1),
+      size = cmapplot_globals$fsize$M/ggplot2::.pt, # pt-to-mm conversion
+      colour = cmapplot_globals$colors$blackish
+    ),
+    RecessionsText = list(
+      family = cmapplot_globals$font$regular$family,
+      fontface = ifelse(cmapplot_globals$font$regular$face == "bold", 2, 1),
+      size = cmapplot_globals$fsize$S/ggplot2::.pt, # pt-to-mm conversion
+      colour = cmapplot_globals$colors$blackish
+    )
+  )
+}
+
+#' Fetch and set aesthetic defaults
+#'
+#' These functions allow for setting and resetting default aesthetic values for
+#' certain ggplot2 geoms. This is necessary for geoms to be "themed" to CMAP
+#' style standards, because (at least at the moment) setting geom aesthetic
+#' defaults on a plot-by-plot basis (such as with \code{ggplot2::theme}) is not
+#' possible. The geoms impacted are stored in
+#' \code{cmapplot_globals$geoms_to_change}.
+#'
+#' These functions are employed implictly within \code{\link{finalize_plot}} to
+#' apply preferred aesthetic defaults to final outputs. They are only necessary
+#' to use explicitly if you would like plots to use these defaults
+#' pre-\code{finalize}.
+#'
+#' CAUTION: Running \code{apply_cmap_default_aes} will set defaults for all
+#' ggplot2 plots drawn in the current session. To reset to \code{ggplot2}
+#' defaults (technically, to whatever the defaults were when \code{cmapplot}
+#' was loaded), use \code{unapply_cmap_default_aes}.
+#'
+#' Note: CMAP aesthetic defaults are loaded into
+#' \code{cmapplot_globals$default_aes_cmap} by the internal pkg function
+#' \code{init_cmap_default_aes} when \code{cmapplot} is first loaded into R.
+#'
+#' @param quietly Should the function suppress confirmation message?
+#'
+#' @examples
+#' \dontrun{
+#' g <- ggplot(filter(grp_over_time, category == "Services"),
+#'             aes(x = year, y = realgrp, color = cluster)) +
+#'   geom_recessions(ymax = 0.4, text_nudge_x = 0.1) +
+#'   theme_cmap(hline = 0,
+#'              axislines = "x",
+#'              legend.max.columns = 2) +
+#'   ggtitle("Change in gross regional product over time") +
+#'   geom_line() +
+#'   scale_x_continuous("Year", breaks = seq(2007, 2017, 2)) +
+#'   coord_cartesian(clip = "off") +
+#' geom_text_lastonly(aes(label = realgrp), add_points = TRUE)
```

Comment:
Are we happy with the default point size for `geom_text_lastonly` after applying `finalize_plot` or `apply_cmap_default_aes`, or should they be bigger? They look a little small to me with the thicker lines, but I'm not sure if we have guidance from Comms on this?

Reply:
Good point. We have models for this

Reply:
I set the default size to 2x the default line width (so, 6 pt?) in commit 2a48ec4ba8ba9a6496322d5488249b2508a6d73e. Looks good enough to me, but we may want to check with Ben at some point.